client/components/domains/use-my-domain/index.jsx

```diff
@@ -422,6 +431,36 @@ function UseMyDomain( props ) {
 		);
 	};
 
+	useEffect( () => {
+		if ( useMyDomainMode && mode !== useMyDomainMode && isStepper ) {
+			setMode( useMyDomainMode );
+		}
+	}, [ useMyDomainMode, mode, isStepper ] );
+
+	useEffect( () => {
+		if ( initialMode ) {
+			setMode( initialMode );
+		}
+	}, [ initialMode ] );
+
+	useEffect( () => {
+		if ( ! initialQuery || initialValidation.current ) {
+			return;
+		}
+
+		initialValidation.current = true;
+		initialQuery &&
```

Comment:
Did not get this part, why do we need it?
`initialQuery && ! initialMode && ! getDomainNameValidationErrorMessage( initialQuery ) && onNext();`

Reply:
This is existing logic. I just moved it to the bottom of the file.